tests/budget_accounting_test.py

```diff
@@ -48,5 +53,147 @@ def test_compute_budgets(self):
         self.assertEqual(budget2.delta, 1e-6)
 
 
+class PLDBudgetAccountantTest(unittest.TestCase):
+
+    def test_noise_not_calculated(self):
+        with self.assertRaises(AssertionError):
+            mechanism = MechanismSpec(NoiseKind.LAPLACE)
+            print(mechanism.noise_standard_deviation())
+
+    def test_invalid_epsilon(self):
+        with self.assertRaises(ValueError):
+            PLDBudgetAccountant(total_epsilon=0, total_delta=1e-5)
+
+    def test_invalid_delta(self):
+        with self.assertRaises(ValueError):
+            PLDBudgetAccountant(total_epsilon=1, total_delta=-1e-5)
+
+    def test_compute_budgets_none_noise(self):
+        accountant = PLDBudgetAccountant(total_epsilon=3, total_delta=1e-5)
+        accountant.compute_budgets()
+        self.assertEqual(None, accountant.minimum_noise_std)
+
+    def test_compute_budgets(self):
+
+        @dataclass
+        class ComputeBudgetMechanisms:
+            count: int
+            expected_noise_std: float
+            noise_kind: NoiseKind
+            weight: float
+            sensitivity: float
+
+        @dataclass
+        class ComputeBudgetTestCase:
+            name: str
+            expected_epsilon: float
+            delta: float
+            expected_pipeline_noise_std: float
+            mechanisms: []
+
+        testcases = [
```

Comment:
@dvadym I thought it might be nice to try table driven tests in Python, but perhaps I got a little carried away and it is too complex, if so let me know and I will revert to the old style. Thank you!

Reply:
Thanks, I like this approach!